lib/Elastica/Processor/Attachment.php

```diff
@@ -44,7 +44,7 @@ public function setTargetField(string $targetField)
      *
      * @return $this
      */
-    public function setIndexedChars(int $indexedChars)
+    public function setIndexedChars(int $indexedChars = 100000): self
```

Comment:
@massimilianobraglia why did you set default value here, and not for example in ```setTargetField``` ?

Reply:
I'll check it later, but probably I just forgot that one :)

Reply:
No problem u're doing a huge *Monster* job!!!
It's really a lot! If you don't mind I'll have a review on default values on 
all classes :) maybe I can be of help

Reply:
Thank you, @p365labs!! 😄
Yep, obviously no problem from me about the review! I'm looking forward that 😄 

I just checked the code and yes, I forgot some default values! I have set those in the latest update!

Reply:
Are these the default values from Elasticsearch? One challenge when we set these is that ES might change them but Elastica does not because we miss it.

Do we need the default? Because I assume if someone calls this method, he does want to change the default?

Reply:
Yep they are the ES default values.

Anyway your's is a good point. From my PoV we have 2 ways:
- using everywhere the default values
- not using everywhere

In my travel through Elastica code, I found places where default values are used (obviously I am not talking about library default values but ES default values). My opinion is to keep them. A use could call that method just to restore the default.

Just a thought 😄

Reply:
@ruflin & @massimilianobraglia I agree with the idea of having default values in Elastica.

Reply:
The challenge I see here is: How do we keep up with defaults that are changing in ES?

From a maintainability perspective I would probably rather prefer not to have defaults in Elastica. I think the ones we have are rather old from the 0.x ages of Elasticsearch. Back then the API was not very extensive and it was still possible to keep up with changes. 

@massimilianobraglia Could you link some default examples that are already there?

Reply:
@ruflin yes no problem :)
I'll link those ASAP

Reply:
@ruflin sorry about the late answer. Anyway, I looked into the codebase of Processor namespace and I couldn't find the default values I was talkin' about. Probably those were in other namespaces (I think).

Should I remove these ones?

Reply:
@massimilianobraglia I would be for not having the defaults but lets' also see if @p365labs agrees :-)

Reply:
I understand the matter about maintainability perspective, and I agree partially. I mean maintaining defaults could be a bit annoying and without a lot of values for a library as Elastica. At the same time I'm wondering if these values could be useful not only in terms of documentation but also in order to restore default behaviours.

Reply:
Maybe we can just add class constants representing the default values, what do you think?
Something like `DEFAULT_* = value`. With these, we don't change any method signature and the user can restore the default value by using that constants

Reply:
@massimilianobraglia I like the middle ground you are proposing. This allows to also only introduce the defaults that are often needed instead of all.

Reply:
ok then! If also @p365labs agrees I will proceed 😄

Reply:
@ruflin @massimilianobraglia totally agree :) sorry for been so late in reply

Reply:
I have just updated this PR. Can you have a look, please?